### PR TITLE
Improvements based on #1122647

### PR DESCRIPTION
**Pull request by mishravikas**
Following improvements has been made in this PR
-Comments can be added to multiple alerts
-Revisions can be changed for one or more than one alerts
-Add bugs to multiple alerts
-Status can be changed for more than one bug together by selecting multiple at once

### CODE REVIEW

File: templates/base.html [hunk 1]
```diff
@@ -112,6 +132,16 @@
           </form>
         </div>
 
+        <div id="changeRevPopup" class="popupForm" style="display: none;">
+          <a class="closePopup" onclick="document.getElementById('changeRevPopup').style.display='none';">close</a>
+          <form id="changeRevForm">
+            <h2>New Revesion</h2>
```
Comment (jmaher) [inline templates/base.html]: typo here, Revision it should be

File: templates/base.html [hunk 2]
```diff
@@ -58,6 +60,25 @@
...
+                                <option>NEW</option>
+                                <option>Back Filling</option>
+                                <option>Investigating</option>
+                                <option>Resolved</option>
+                                <option>Shipped</option>
+                                <option>Wont Fix</option>
+                                <option>False Alarm</option>
+                                <option>Ignore</option>
+                                <option>Not Tracking</option>
+                                <option>Too Low</option>
```
Comment (jmaher) [inline templates/base.html]: will these options change the status for us automatically?

File: templates/alerts.html [hunk 3]
```diff
@@ -22,13 +22,15 @@
...
+            rev.className="check-alert";
+            rev.innerHTML="<input type=\"checkbox\" id =\"check-"+alert['id']+"\"/>";
 
             for (var i = 0; i < fields.length; i++) {
                 var test = fields[i];
                 cell = row.insertCell(-1);
                 cell.className = cname;
                 cell.setAttribute('field_name', fields[i]);
-                cell.onclick = editAlert(alert["id"], alert["body"]);
+                // cell.onclick = editAlert(alert["id"], alert["body"]);
```
Comment (jmaher) [inline templates/alerts.html]: is there a reason for this?  I assume this gets rid of the popup whenever you click on the alert row.  I would prefer to remove the line entirely then!

File: server.py [hunk 4]
```diff
@@ -14,6 +14,7 @@
 def home():
     return render_template('alerts.html')
 
+
```
Comment (jmaher) [inline server.py]: nit: extra blank line here, I don't think we need it.

File: js/alerts.js [hunk 5]
```diff
@@ -196,6 +197,146 @@ function updateStatus(alertid, duplicate, bugid, mergedfrom) {
...
+                    });
+
+                }
+                location.reload();
+            
+            }
+        }
+
+        else if(action == "Add Comment"){
+            console.log("REACHED Add Comment")
```
Comment (jmaher) [inline js/alerts.js]: this looks like debugging.

File: templates/base.html (hunk 2 above)
Comment (mishravikas) [inline templates/base.html]: yeah they will change automatically and are similar to changing status for each alert by selecting options from that particular row

File: templates/alerts.html (hunk 3 above)
Comment (mishravikas) [inline templates/alerts.html]: Yes this gets rid of the popup, I'll remove it completely then

File: js/alerts.js (hunk 5 above)
Comment (mishravikas) [inline js/alerts.js]: you're talking about the console.log() line?

Comment (jmaher) [inline js/alerts.js]: yes!  just the console.log line

File: js/alerts.js [hunk 6]
```diff
@@ -196,6 +197,143 @@ function updateStatus(alertid, duplicate, bugid, mergedfrom) {
...
+    //get ids of all the checked alerts
+   var checkedIds = $(":checkbox:checked").map(function() {
+        return this.id;
+    }).get();
+   //Check if atleast one alert is chosen
+   if (checkedIds.length<=0 && action != "Actions"){
+    console.log("No alert chosen");
+    alert("Please Choose atleast one Alert");
+   }
+   else{
```
Comment (jmaher) [inline js/alerts.js]: nit: } else {

also please always add a space between ){

File: js/alerts.js [hunk 7]
```diff
@@ -196,6 +197,143 @@ function updateStatus(alertid, duplicate, bugid, mergedfrom) {
...
+                    data: {
+                        id: checkedIds[id],
+                        status: action,
+                    }
+                });
+            }
+            location.reload();
+        }
+
+        else if(action == "Change Revision"){
```
Comment (jmaher) [inline js/alerts.js]: when we update the revision there is a good change we will need to update the branch as well- a follow up PR/bug could be to make this more streamlined as well as updating the tbplurl generated link.

File: js/alerts.js [hunk 8]
```diff
@@ -196,6 +197,143 @@ function updateStatus(alertid, duplicate, bugid, mergedfrom) {
...
+                    buttons: { 
+                        Ok: function() {
+                            console.log($("#commentName").val());
+                            console.log($("#commentText").val());
+                            var email = $("#commentName").val();
+                            var comment = $("#commentText").val();
+                            console.log(typeof(email)+typeof(comment));
+                            for (id in checkedIds){
+                                    console.log(checkedIds[id]);
+                                $.ajax({
```
Comment (jmaher) [inline js/alerts.js]: nit: spacing of the $.ajax({ line, this should be indented 4 more spaces

File: server.py [hunk 9]
```diff
@@ -216,6 +220,53 @@ def run_submit_data():
...
+    To Add comments, bugs to multiple alerts
+    TODO: implemented only comments as of now
+
+    '''
+    sql=''
+    query_dict = request.args.to_dict()
+    typeVal = ""
+    data = request.form
+    retVal = {}
+    data = request.form
```
Comment (jmaher) [inline server.py]: why do you have data = request.form twice here?

Comment (mishravikas) [inline server.py]: oops! will remove that

File: js/alerts.js (hunk 7 above)
Comment (mishravikas) [inline js/alerts.js]: allright then will work on this next

File: js/alerts.js (hunk 8 above)
Comment (mishravikas) [inline js/alerts.js]: I think L309 should be indented 4 less spaces instead

Comment (jmaher) [inline js/alerts.js]: yes, you are right.

File: js/alerts.js [hunk 10]
```diff
@@ -169,6 +169,7 @@ function addMergedLinks(showall) {
 
 function updateStatus(alertid, duplicate, bugid, mergedfrom) {
     var status = $(document.getElementById(alertid + "-status")).val();
+    console.log(status);
```
Comment (jmaher) [inline js/alerts.js]: this could be more descriptive or should be removed.

File: js/alerts.js [hunk 11]
```diff
@@ -196,6 +197,143 @@ function updateStatus(alertid, duplicate, bugid, mergedfrom) {
     }
 }
 
+
+//function to check whether a name exists in an array
+function containsObject(obj, list) {
+    var i;
+    for (i = 0; i < list.length; i++) {
```
Comment (jmaher) [inline js/alerts.js]: for (var i = 0...

File: js/alerts.js [hunk 12]
```diff
@@ -196,6 +197,143 @@ function updateStatus(alertid, duplicate, bugid, mergedfrom) {
...
+    //get ids of all the checked alerts
+   var checkedIds = $(":checkbox:checked").map(function() {
+        return this.id;
+    }).get();
+   //Check if atleast one alert is chosen
+   if (checkedIds.length<=0 && action != "Actions") {
+    console.log("No alert chosen");
+    alert("Please Choose atleast one Alert");
+   }
+   else {
```
Comment (jmaher) [inline js/alerts.js]: } else {

File: js/alerts.js [hunk 13]
```diff
@@ -196,6 +197,143 @@ function updateStatus(alertid, duplicate, bugid, mergedfrom) {
...
+
+        for (alert in checkedIds) {
+            checkedIds[alert] = checkedIds[alert].split('-')[1];
+        }
+        console.log(checkedIds);
+        //Check if status has to be changed
+        if (containsObject(action, status_options)) {
+            console.log("change status-"+action);
+            //Update the status one by one by calling updatestatus()
+            for (id in checkedIds){
```
Comment (jmaher) [inline js/alerts.js]: space between ){ <- please do that to the entire patch

File: js/alerts.js [hunk 14]
```diff
@@ -196,6 +197,143 @@ function updateStatus(alertid, duplicate, bugid, mergedfrom) {
...
+                    data: {
+                        id: checkedIds[id],
+                        status: action,
+                    }
+                });
+            }
+            location.reload();
+        }
+
+        else if(action == "Change Revision") {
```
Comment (jmaher) [inline js/alerts.js]: space between if(

File: js/alerts.js [hunk 15]
```diff
@@ -196,6 +197,143 @@ function updateStatus(alertid, duplicate, bugid, mergedfrom) {
...
+            var newRev = prompt("Please enter new Revision");
+            for (id in checkedIds) {
+
+                var bug = '1234';
+                if (bug == '') {
+                    bug = $(document.getElementById(alertid + "-bug")).val();
+                }
+    
+                if (newRev != null) {
+                    console.log('new rev-'+newRev);
```
Comment (jmaher) [inline js/alerts.js]: these console.log messages should be much more descriptive or removed.

File: js/alerts.js [hunk 16]
```diff
@@ -196,6 +197,143 @@ function updateStatus(alertid, duplicate, bugid, mergedfrom) {
...
+            }
+            location.reload();
+
+        }
+
+        else if(action == "Add Bug") {
+            var BugID = prompt("Please enter Bug ID");
+            console.log('BUG ID-'+BugID);
+            if (BugID != null) {
+                for(id in checkedIds){
```
Comment (jmaher) [inline js/alerts.js]: space between for( and ){

File: js/alerts.js [hunk 17]
```diff
@@ -196,6 +197,143 @@ function updateStatus(alertid, duplicate, bugid, mergedfrom) {
...
+                    }
+                });
+            });
+            $("#addCommentpopup").dialog("open");
+
+        }   
+
+   }
+   
+}
```
Comment (jmaher) [inline js/alerts.js]: there are 3 blank lines here, lets just have:
    }
  }
}

**Comment by jmaher** [comment]
please let me know when you have addressed the comments, I can look and merge.  Going forward, I would like to fix up the branch/revision updating so we can resolve the tbplurl (generated in parse_news.py, we will need to find a way to call that function from server.py).  Possibly update graphurl as well.

**Pull request merged**